packages/peering/lpp/lppNetImpl.go

```diff
@@ -0,0 +1,549 @@
+			fmt.Sprintf("/ip6/::1/tcp/%v", port),
+		),
+		libp2p.Transport(tcp.NewTCPTransport),
+		libp2p.Transport(libp2pquic.NewTransport),
+	)
+	if err != nil {
+		ctxCancel()
+		return nil, nil, xerrors.Errorf("failed to construct libp2p host: %w", err)
+	}
+	n := netImpl{
+		myNetID:     myNetID,
+		lppHost:     lppHost,
+		ctx:         ctx,
+		ctxCancel:   ctxCancel,
+		port:        port,
+		peers:       make(map[libp2ppeer.ID]*peer),
+		peersLock:   &sync.RWMutex{},
+		recvEvents:  nil, // Initialized bellow.
+		recvQueue:   make(chan *peering.RecvEvent, recvQueueSize),
+		nodeKeyPair: nodeKeyPair,
+		trusted:     trusted,
+		log:         log,
+	}
+	n.recvEvents = events.NewEvent(n.eventHandler)
+	//
+	// Finish initialization of the libp2p node.
+	lppHost.SetStreamHandler(lppProtocolPeering, n.lppPeeringProtocolHandler)
+	lppHost.SetStreamHandler(lppProtocolHeartbeat, n.lppHeartbeatProtocolHandler)
+	trustedPeers, err := trusted.TrustedPeers()
+	if err != nil {
+		ctxCancel()
+		return nil, nil, xerrors.Errorf("unable to get trusted peers: %w", err)
+	}
+	for _, trustedPeer := range trustedPeers {
+		if err := n.addPeer(trustedPeer); err != nil {
+			ctxCancel()
+			return nil, nil, xerrors.Errorf("unable to setup trusted peer: %w", err)
+		}
+	}
+	return &n, &n, nil
+}
+
+func (n *netImpl) lppAddToPeerStore(trustedPeer *peering.TrustedPeer) (libp2ppeer.ID, error) {
+	lppPeerID, lppPeerPub, err := n.lppTrustedPeerID(trustedPeer)
+	if err != nil {
+		return libp2ppeer.ID(""), err
+	}
+	peerHost, peerPort, err := peering.ParseNetID(trustedPeer.NetID)
+	if err != nil {
+		return libp2ppeer.ID(""), xerrors.Errorf("failed to parse trusted peer NetID=%v, error: %w", trustedPeer.NetID, err)
+	}
+	//
+	// Resolve IP addresses.
+	peerIP4 := make([]string, 0)
+	peerIP6 := make([]string, 0)
+	peerIPs, err := net.LookupIP(peerHost)
+	if err != nil {
+		return libp2ppeer.ID(""), xerrors.Errorf("failed to lookup IPs for NetID=%v, error: %w", trustedPeer.NetID, err)
+	}
+	for i := range peerIPs {
+		if ip4 := peerIPs[i].To4(); ip4 != nil {
+			peerIP4 = append(peerIP4, ip4.String())
+		} else {
+			peerIP6 = append(peerIP6, peerIPs[i].String())
+		}
+	}
+	//
+	// Create multiaddresses.
+	addrPatterns := []string{
+		"/%s/%s/udp/%v/quic",
+		"/%s/%s/tcp/%v",
+	}
+	addrs := make([]multiaddr.Multiaddr, 0)
+	for i := range addrPatterns {
+		for j := range peerIP4 {
+			addr, err := multiaddr.NewMultiaddr(fmt.Sprintf(addrPatterns[i], "ip4", peerIP4[j], peerPort))
+			if err != nil {
+				return libp2ppeer.ID(""), xerrors.Errorf("failed to make libp2p address for NetID=%v, error: %w", trustedPeer.NetID, err)
+			}
+			addrs = append(addrs, addr)
+		}
+		for j := range peerIP6 {
+			addr, err := multiaddr.NewMultiaddr(fmt.Sprintf(addrPatterns[i], "ip6", peerIP6[j], peerPort))
+			if err != nil {
+				return libp2ppeer.ID(""), xerrors.Errorf("failed to make libp2p address for NetID=%v, error: %w", trustedPeer.NetID, err)
+			}
+			addrs = append(addrs, addr)
+		}
+	}
```

Comment:
seems a bit of duplicated logic here (for ip4/6), would it make sense to do something like this? (and we could remove the ip4/ip6 lists split)


```suggestion
func getIpVersion( ip IP) string {
	if ip.To4() != nil {
		return "ip4"
	}
	return "ip6"
}

// ---------
		for _, ip := range peerIPs {
			addr, err := multiaddr.NewMultiaddr(fmt.Sprintf(addrPatterns[i], getIpVersion(ip),ip, peerPort))
			if err != nil {
				return libp2ppeer.ID(""), xerrors.Errorf("failed to make libp2p address for NetID=%v, error: %w", trustedPeer.NetID, err)
			}
			addrs = append(addrs, addr)
		}
	}
```

Reply:
Updated.